static/js/src/advantage/subscribe/react/APICalls/usePostCustomerInfoForPurchasePreview.jsx

```diff
@@ -22,28 +22,24 @@ const usePostCustomerInfoAnon = () => {
       state: country === "US" ? usState : caProvince,
     };
 
-    if (VATNumber) {
-      const res = await postCustomerInfoForPurchasePreview(
-        window.accountId,
-        name,
-        addressObject,
-        {
-          type: "eu_vat",
-          value: VATNumber,
-        }
-      );
-
-      if (res.errors) {
-        throw new Error(JSON.parse(res.errors).code);
+    const res = await postCustomerInfoForPurchasePreview(
+      window.accountId,
+      name,
+      addressObject,
+      {
+        type: "eu_vat",
+        value: VATNumber,
       }
+    );
 
-      return res;
+    if (res.errors) {
+      throw new Error(JSON.parse(res.errors).code);
     }
 
-    throw new Error("VAT is missing");
```

Comment:
I found this to be a little confusing, the API does not return an error when VAT number is missing. This seems like a perfectly valid call to make (as explained above).

Reply:
Yeah the reason I added that was to make an effort to not call the API when we know it's not going to do anything. The condition should probably have been VATApplicableCountry && vatNumber
But I guess it's fine to let it go through as well

Reply:
Correct me if I'm wrong, but If it was `VATApplicableCountry && vatNumber` the issue would still occur. Looks like we always need to make 2 calls (postCustomerInfoForPurchasePreview and then preview) if any data that could affect the preview price has changed (including if we switched to a non-vat-applicable country). Otherwise preview will return response based on stale form data.